src/paas_charm/charm.py

```diff
@@ -481,6 +514,12 @@ def _create_charm_state(self) -> CharmState:
                 "service_name": self.app.name,
                 "endpoint": f"{self._tracing.get_endpoint(protocol='otlp_http')}",
             }
+        smtp_relation_data = None
+        if self._smtp and (smtp_data := self._smtp.get_relation_data()):
+            smtp_relation_data = smtp_data.to_relation_data()
+            if password_id := smtp_relation_data.get("password_id"):
+                secret = self.model.get_secret(id=password_id)
+                smtp_relation_data["password"] = secret.get_content()
```

Comment:
This should be provided by the library, when you're doing "get_relation_data" the library should extract the password value, and directly provide it in the pydantic object.
I think we should push that change in the library